iml-gui/crate/src/lib.rs

```diff
@@ -319,8 +343,12 @@ pub fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg, GMsg>)
             orders.send_msg(Msg::LoadPage);
         }
         Msg::UpdatePageTitle => {
-            let title = format!("{} - {}", model.page.title(), TITLE_SUFFIX);
-            document().set_title(&title);
+            let title = model.page.title();
+            document().set_title(&format!("{} - {}", &title, TITLE_SUFFIX));
+            model.breadcrumbs.push(BreadCrumb {
```

Comment:
I wonder if we should send a message to breadcrumbs here wit hthe BreadCrumb object and let the update function inside of breadcrumbs handle the insertion?

Reply:
It looks complicated.

Reply:
I might be reading it incorrectly. I was thinking since you have the breadcrumbs component, wouldn't it make sense to maintain the state inside of that component? And if the state is maintained inside of the component, shouldn't we send messages to the component whenever the state should be updated?

Reply:
I think messages should be used (inevitably) when we are dealing with async operations. And this is not the case. Breadcrumbs are just a set of helper functions.

Reply:
Perhaps it is more about aesthetics, as what you have clearly works. It just seems cleaner to have the component update its own model instead of a parent updating a child model. In other words:
```
Msg::Action => child.model.count += 1;
```
vs
```
Msg::Action => {
   orders.proxy(Msg::Child).send_msg(child::Msg::Increment);
}
Msg::Child(msg) => {
   child::update(msg, &mut model.child, &mut orders.proxy(Msg::Child));
}
```
In my opinion, the advantage of the second method is that the parent is not responsible for the update logic. Instead, the message is proxied to the appropriate component and handled there.

Reply:
I think this is a nit, but I agree with @johnsonw here. As the app grows larger, it's easier to reason about if a section's model is only mutated within it's own update fn (especially if it has one).